scripts/translate_lessons.py

```
@@ -0,0 +1,318 @@
+    # Per-language cache so parallel language jobs never touch the same file
+    # (no push races) and a timed-out run resumes exactly where it stopped.
+    return OUT_ROOT / lang / ".translate-cache.json"
+
+def _load_registry():
+    # languages.json is a committed canonical file; fail loudly if it is missing
+    # rather than masking that with a silent hardcoded fallback.
+    return json.loads((ROOT / "languages.json").read_text(encoding="utf-8"))["languages"]
+
+
+_REG = _load_registry()
+LANG_NAMES = {l["code"]: l["name"] for l in _REG if not l.get("source")}
```

Ruff (0.16.0)

[error] Ambiguous variable name: `l`

(E741)